app/stac_api/validators.py

```diff
@@ -216,3 +223,306 @@ def validate_asset_multihash(value):
             message=_('Invalid multihash value; %(error)s'),
             params={'error': error}
         )
+
+
+class ValidateSearchRequest:
+    '''Validates the query parameter for the search endpoint.
+
+    The main function is validate. If everything in the query parameter is ok
+    this class does nothing.
+
+    If there are errors in the validation it sums them up and returns the summary
+    when raising a ValidationError.
+    '''
+
+    def __init__(self):
+        self.errors = {}  # a list with all the validation errors
+        self.max_len_array = 2000
+        self.max_times_same_query_attribute = 20
+        self.max_query_attributes = 50
+
+        self.queriabel_date_fields = ['datetime', 'created', 'updated']
+        self.queriabel_str_fields = ['title']
+
+    def validate(self, request):
+        '''Validates the request of the search endpoint
+
+        This function validates the request of the search endpoint. As a simplification the
+        requests of GET and POST are harmonized. Then the search params are validated. This
+        function gathers as much validation information as possible. If there is one error
+        or several, finally it raises one error with a complete validation feedback.
+
+        Args:
+            request (RequestDict)
+                The Request (POST or GET)
+
+        Raises:
+            ValidationError(code, details)
+        '''
+        # harmonize GET and POST
+        query_param = harmonize_post_get_for_search(request)
+
+        if 'bbox' in query_param:
+            self.validate_bbox(query_param['bbox'])
+        if 'datetime' in query_param:
+            self.validate_date_time(query_param['datetime'])
+        if 'ids' in query_param:
+            self.validate_array_of_strings(query_param['ids'], 'ids')
+        if 'collections' in query_param:
+            self.validate_array_of_strings(query_param['collections'], 'collections')
+        if 'query' in query_param:
+            self.validate_query(query_param['query'])
+        if 'intersects' in query_param:  # only in POST
+            self.validate_intersects(json.dumps(query_param['intersects']))
+
+        # Raise ERROR with a list of parsed errors
+        if self.errors:
+            for key in self.errors:
+                logger.error('%s: %s', key, self.errors[key])
+            raise RestValidationError(code='query-invalid', detail=self.errors)
+
+    def validate_query(self, query):
+        '''Validates the query parameter
+
+        The query parameter is being validated. If an validation error is detected,
+        an information is being added to the dict self.errors
+
+        Args:
+            query: string
+                The query parameter to be validated.
+        '''
+        # summing up the fields based of different types
+        queriabel_fields = self.queriabel_date_fields + self.queriabel_str_fields
+
+        # validate json
+        try:
+            json_query = json.loads(query)
```

Comment:
being picky: this is rather `query_dict` than a `json_query` (the object is a dict, not a string formatted as json)... so it's rather `query_dict = json.loads(json_query)`...